examples/qml_features/rust/src/signals.rs

```diff
@@ -49,6 +63,57 @@ pub mod ffi {
             // Emit a signal to QML stating that we have disconnected
             self.as_mut().emit(Connection::Disconnected);
         }
+
+        #[qinvokable]
+        pub fn initialise(self: Pin<&mut Self>) {
+            self.on_logging_enabled_changed(
+                |mut qobject| {
+                    // Determine if logging is enabled
+                    if *qobject.as_ref().logging_enabled() {
+                        // If no connections have been made, then create them
+                        if qobject.as_ref().connections().is_none() {
+                            // ANCHOR: book_signals_connect
+                            let connections = [
+                                qobject.as_mut().on_connected(
+                                    |_, url| {
+                                        println!("Connected: {}", url);
+                                    },
+                                    ConnectionType::AutoConnection,
+                                ),
+                                qobject.as_mut().on_disconnected(
+                                    |_| {
+                                        println!("Disconnected");
+                                    },
+                                    ConnectionType::AutoConnection,
+                                ),
+                                qobject.as_mut().on_error(
+                                    |_, message| {
+                                        println!("Error: {}", message);
+                                    },
+                                    ConnectionType::AutoConnection,
+                                ),
+                            ];
+                            qobject.as_mut().set_connections(Some(connections));
+                            // ANCHOR_END: book_signals_connect
+                        }
+                    } else {
+                        // Disabling logging so disconnect
+                        // ANCHOR: book_signals_disconnect
+                        // Could also empty the connections which would cause drop to trigger a disconnect
+                        // qobject.as_mut().set_connections(None);
+                        if let Some(connections) = qobject.as_mut().connections_mut().take() {
+                            for conn in connections {
+                                conn.disconnect();
+                            }
+                        }
+                        // ANCHOR_END: book_signals_disconnect
+                    }
+                },
+                ConnectionType::AutoConnection,
+            )
+            .pin_mut()
```

Comment:
Need to get from `UniquePtr<QMetaObjectConnectionGuard>` to `Pin<&mut QMetaObjectConnectionGuard>` which doesn't happen via deref to be able to reach the non-const `release` method.

The other option would be to make inner `mutable` in the C++ object and then make `release` const. But maybe forcing the `pin_mut` is fine?

Reply:
Changed to trivial.